editor/editor_resource_picker.cpp

```diff
@@ -227,17 +242,16 @@ void EditorResourcePicker::_update_menu_items() {
 	// Add options to copy/paste resource.
 	Ref<Resource> cb = EditorSettings::get_singleton()->get_resource_clipboard();
 	bool paste_valid = false;
-	if (is_editable()) {
-		if (cb.is_valid()) {
-			if (base_type.is_empty()) {
-				paste_valid = true;
-			} else {
-				for (int i = 0; i < base_type.get_slice_count(","); i++) {
-					if (ClassDB::is_parent_class(cb->get_class(), base_type.get_slice(",", i))) {
-						paste_valid = true;
-						break;
-					}
-				}
+	if (is_editable() && cb.is_valid()) {
+		if (base_type.is_empty()) {
+			paste_valid = true;
+		} else {
+			String res_type = _get_resource_type(cb);
+
+			for (int i = 0; i < base_type.get_slice_count(",") && !paste_valid; i++) {
+				String base = base_type.get_slice(",", i);
+
+				paste_valid = ClassDB::is_parent_class(cb->get_class(), base) || EditorNode::get_editor_data().script_class_is_parent(res_type, base);
```

Comment:
Should break on hit as well.

Reply:
Also, shouldn't the ClassDB check also use `res_type`, if for consistency at least?